spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/SparkSizeBasedDataRewriter.java

```diff
@@ -57,7 +64,34 @@ public Set<DataFile> rewrite(List<FileScanTask> group) {
 
       doRewrite(groupId, group);
 
-      return coordinator.fetchNewFiles(table(), groupId);
+      Set<DataFile> newFiles = coordinator.fetchNewFiles(table(), groupId);
+
+      PartitionData partition = (PartitionData) group.get(0).partition();
```

Comment:
Is it safe to assume about the type `PartitionData`, without type checking explicitly?

Reply:
Probably, but do you need to?